curriculum/challenges/english/25-front-end-development/lab-user-configuration-manager/684aaf9ec670c68d20efd0d0.md

```diff
@@ -0,0 +1,456 @@
+
+```js
+({
+    test: () => runPython(`
+        settings_in_test = {"theme": "light", "language": "English"}
+        result = delete_setting(settings_in_test, "theme")
+        expected_msg = "Setting 'theme' deleted successfully!"
+        assert result == expected_msg
+        assert "theme" not in settings_in_test
+    `)
+})
+```
+
+`delete_setting` should return the error message `Setting not found!` when the key doesn't exist.
+
+```js
+({
+    test: () => runPython(`
+        settings_in_test = {"theme": "light"}
+        result = delete_setting(settings_in_test, "volume")
+        expected_msg = "Setting not found!"
+        assert result == expected_msg
+    `)
+})
+```
+
+`delete_setting` should correctly remove the given key from the dictionary.
+
+```js
+({
+    test: () => runPython(`
+        settings_in_test = {"theme": "light", "language": "English"}
+        result = delete_setting(settings_in_test, "language")
+        assert "language" not in settings_in_test
+    `)
+})
+```
+
+You should define a function named `view_settings`.
+
+```js
+({
+    test: () => runPython(`
+        assert _Node(_code).has_function('view_settings')
+    `)
+})
+```
+
+The `view_settings` function should take one parameter.
+
+```js
+({
+    test: () => runPython(`
+        import inspect
+        sig = inspect.signature(view_settings)
+        assert len(sig.parameters) == 1
+    `)
+})
+```
+
+`view_settings` should return the message `No settings available.` if the given dictionary is empty.
+
+```js
+({
+    test: () => runPython(`
+        empty_settings = {}
+        result = view_settings(empty_settings)
+        expected_msg = "No settings available."
+        assert result == expected_msg
+    `)
+})
+```
+
+`view_settings` should return formatted settings for non-empty dictionary.
+
+```js
+({
+    test: () => runPython(`
+        settings_in_test = {"theme": "dark", "notifications": "enabled", "volume": "high"}
+        result = view_settings(settings_in_test)
+        assert result.startswith("Current User Settings:")
+        assert "Theme: dark" in result
+        assert "Notifications: enabled" in result
+        assert "Volume: high" in result
+    `)
+})
+```
+
+`view_settings` should capitalize the first letter of each setting name.
+
+```js
+({
+    test: () => runPython(`
+        settings_in_test = {"theme": "dark", "volume": "high"}
+        result = view_settings(settings_in_test)
+        assert "Theme:" in result
+        assert "Volume:" in result
+        assert "theme:" not in result
+        assert "volume:" not in result
+    `)
+})
+```
+
+`view_settings` should display the correct results.
+
+```js
+({
+    test: () => runPython(`
+        settings_in_test = {"theme": "light", "language": "English", "notifications": "enabled"}
+        result = view_settings(settings_in_test)
+        assert result == "Current User Settings:\\nTheme: light\\nLanguage: English\\nNotifications: enabled\\n"
```

Comment:
The final newline character doesn't seem to be included in the example output. We should either remove it or say explicitly that is required.